src/Http/Middleware/XssProtectionHeader.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Http\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class XssProtectionHeader implements Middleware
+{
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $response = $handler->handle($request);
+
+        return $response->withAddedHeader('X-XSS-Protection', '1; mode=block');
```

Comment:
How does this affect custom JS added in the head element, or via custom header / footer? Or the custom JS used in TextFormatter for code highlighting and media embed, for that matter?

Reply:
This header is deprecated and only supported in IE.

Chrome disabled all effects of this header from 77, and FF never implemented it in the first place.

Source: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection

Reply:
It won't block any of the JS added via the header or via embeds as far as I've tested. To be honest I'm not 100% honest how the X-XSS-Protection header works but I do know that it doesn't seem to affect anything on my own forum where I've added it via my webserver.

Reply:
Maybe we shouldn't include it, and focus efforts on supporting CSP instead? That seems to be the more modernly accepted option for doing what we want to do here.